src/tools/tidy/src/style.rs

```diff
@@ -136,15 +137,28 @@ macro_rules! suppressible_tidy_err {
     };
 }
 
+pub fn is_in(full_path: &Path, folder_to_find: &str) -> bool {
```

Comment:
Should we stop iterating on the rust project directory?
```rust
let mut full_path = full_path;
while let Some(parent) = full_path.parent() {
    if let Some(f) = parent.file_name() {
        if f == folder_to_find {
            return true;
        } else if f == RUST_DIR {
            return false;
        } 
    }
    full_path = parent;
}
false
```
Just an example, it could be shorten.

Reply:
What is `RUST_DIR` in this case?

Reply:
Where rust repository is, I don't know what should it be, is there something like this? I think yours is simpler but it may hit something else created by the user.

Reply:
Well, it's very unlikely but I can add a check on the parent as well (it might actually be a good idea).